src/java/main/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -222,10 +224,10 @@ public boolean isQuorumSynced(QuorumVerifier qv) {
     Leader(QuorumPeer self,LeaderZooKeeperServer zk) throws IOException {
         this.self = self;
         try {
+            ServerSocketChannel ssc = ServerSocketChannel.open();
```

Comment:
These changes seem minimal, but this is changing a very low level part of communication so there are potentially compounding effects. I realize we will test this internally, but I do have concerns. Are there any relevant discussions on the zookeeper mailing lists or upstream issue tracker?